### PR TITLE
Replace readSizeType() method and read the index file directly

### PR DESCRIPTION
**Pull request by TrustHenry**
Replace readSizeType() method and read the index file directly
~Dependent on #354~

### CODE REVIEW

File: source/agora/node/BlockStorage.d [hunk 1]
```diff
@@ -786,11 +786,18 @@ public class BlockStorage : IBlockStorage
...
+                    ubyte[] res;
+                    res.length = size;
+                    idx_file.rawRead(res);
+                    return res;
+                };
+                deserializePart(height, dg, CompactMode.No);
 
                 idx_file.rawRead(hash);
 
-                pos = idx_file.readSizeType();
```
Comment (MichaelKim20) [inline source/agora/node/BlockStorage.d:793]: Please, remove unused function `readSizeType`

Comment (TrustHenry) [reply source/agora/node/BlockStorage.d:793]: removed function `readSizeType`.

**Review by MichaelKim20 — APPROVED** [review]

**Comment by Geod24** [comment]
@TrustHenry : I suggest to make this PR independent from #354 by removing the first commit.
Once this is merged, #354 can be rebased on top of `v0.x.x`

**Comment by TrustHenry** [comment]
> @TrustHenry : I suggest to make this PR independent from #354 by removing the first commit.
> Once this is merged, #354 can be rebased on top of `v0.x.x`

I made one commit.

File: source/agora/node/BlockStorage.d [hunk 2]
```diff
@@ -784,13 +784,22 @@ public class BlockStorage : IBlockStorage
...
+                ubyte[] res;
+                res.length = size;
+                idx_file.rawRead(res);
+                return res;
+            };
+
             foreach (idx; 0 .. record_count)
             {
-                height = idx_file.readSizeType();
+                deserializePart(height, dg, CompactMode.No);
```
Comment (Geod24) [inline source/agora/node/BlockStorage.d]: That's not going to compile, you need to remove the third argument.

Comment (TrustHenry) [reply source/agora/node/BlockStorage.d]: The compact size will be applied to #354.

**Comment by AndrejMitrovic** [comment]
> The compact size will be applied to #354.

I think what @Geod24 suggested is to first write this code as `deserializePart(height, dg);`, this PR can then be merged. Then, in #354 you can change it to `deserializePart(height, dg, CompactMode.No);`

This way we can merge #358 right away, and #354 can be merged after @Geod24 reviews and approves it.

File: source/agora/node/BlockStorage.d [hunk 3]
```diff
@@ -784,13 +784,22 @@ public class BlockStorage : IBlockStorage
 
             size_t record_size = (size_t.sizeof * 2 + Hash.sizeof);
             size_t record_count = idx_file.size / record_size;
+
```
Comment (AndrejMitrovic) [inline source/agora/node/BlockStorage.d:787]: Could you reword the commit message a bit?

> Replace readSizeType() method and read the index file directly

Comment (TrustHenry) [reply source/agora/node/BlockStorage.d:787]: Fixed.

**Review by AndrejMitrovic — APPROVED** [review]

**Pull request merged**